### PR TITLE
Add warning commands

### PR DESCRIPTION
**Pull request by iiiiii1wepfj**
~todo: add /setwarnslimit~ done

### CODE REVIEW

**Review by AndrielFR — CHANGES_REQUESTED** [review]
Some commands are missing, such as:

- /warns

The warns should support text message to let us know the reason for the warn.
Below are some comments, look at them.

File: plugins/warns.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    add_warns(m.chat.id, target_user.id, 1)
+    user_warns = get_warns(m.chat.id, target_user.id)
+    if user_warns >= warns_limit:
+        await c.kick_chat_member(m.chat.id, target_user.id)
+        await m.reply_text(f"the user {target_user.mention} was banned because he was warned {user_warns} of {warns_limit} times")
+    else:
+        await m.reply(f"the user {target_user.mention} has {user_warns} of {warns_limit} warnings")
+        
+        
+@Client.on_message(filters.command("unwarn", prefix) & filters.group)
```
Comment (AndrielFR) [inline plugins/warns.py:55]: It should be `/resetwarns` instead of `/unwarn`.

File: plugins/warns.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+@Client.on_message(filters.command("warn", prefix) & filters.group)
+@require_admin(permissions=["can_restrict_members"])
+async def warn_user(c: Client, m: Message):
+    target_user = await get_target_user(c, m)
+    warns_limit = 3
+    add_warns(m.chat.id, target_user.id, 1)
+    user_warns = get_warns(m.chat.id, target_user.id)
+    if user_warns >= warns_limit:
+        await c.kick_chat_member(m.chat.id, target_user.id)
+        await m.reply_text(f"the user {target_user.mention} was banned because he was warned {user_warns} of {warns_limit} times")
```
Comment (AndrielFR) [inline plugins/warns.py:50]: The bot is multi-language, it doesn't make sense that it's just English.

File: plugins/warns.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    else:
+        await m.reply(f"the user {target_user.mention} has {user_warns} of {warns_limit} warnings")
```
Comment (AndrielFR) [inline plugins/warns.py:52]: The bot is multi-language, it doesn't make sense that it's just English.

File: plugins/warns.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        
+        
+@Client.on_message(filters.command("unwarn", prefix) & filters.group)
+@require_admin(permissions=["can_restrict_members"])
+async def unwarn_user(c: Client, m: Message):
+    target_user = await get_target_user(c, m)
+    reset_warns(m.chat.id, target_user.id)
+    await m.reply_text(f"the warns of the user {target_user.mention} was removed")
```
Comment (AndrielFR) [inline plugins/warns.py:60]: The bot is multi-language, it doesn't make sense that it's just English.

File: plugins/warns.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+                       (number, chat_id, user_id))
+        db.commit()
+    else:
+        dbc.execute('INSERT INTO user_warns (user_id, chat_id, count) VALUES (?,?,?)', (user_id, chat_id, number))
+        db.commit()
+    return True
+
+
+def set_warns_limit(chat_id, rules):
+    cursor.execute('UPDATE chats SET rules = ? WHERE chat_id = ?', (rules, chat_id))
```
Comment (AndrielFR) [inline plugins/warns.py:32]: I didn't understand it here.

File: plugins/warns.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    conn.commit()
+
+
+def reset_warns(chat_id, user_id):
+    dbc.execute('DELETE FROM user_warns WHERE chat_id = ? AND user_id = ?', (chat_id, user_id))
+    db.commit()
+    return True
```
Comment (AndrielFR) [inline plugins/warns.py:39]: It does not make sense to return `True` if there is no error checking.

File: plugins/warns.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+def add_warns(chat_id, user_id, number):
+    dbc.execute('SELECT * FROM user_warns WHERE chat_id = ? AND user_id = ?', (chat_id, user_id))
+    if dbc.fetchone():
+        dbc.execute('UPDATE user_warns SET count = count + ? WHERE chat_id = ? AND user_id = ?',
+                       (number, chat_id, user_id))
+        db.commit()
+    else:
+        dbc.execute('INSERT INTO user_warns (user_id, chat_id, count) VALUES (?,?,?)', (user_id, chat_id, number))
+        db.commit()
+    return True
```
Comment (AndrielFR) [inline plugins/warns.py:28]: It does not make sense to return `True` if there is no error checking.

Comment (iiiiii1wepfj) [reply plugins/warns.py:28]: this is from the old bot

File: plugins/warns.py (hunk 6 above)
Comment (iiiiii1wepfj) [reply plugins/warns.py:39]: this is from the old bot

File: plugins/warns.py (hunk 4 above)
Comment (iiiiii1wepfj) [reply plugins/warns.py:60]: after this we add a strings

Comment (AndrielFR) [reply plugins/warns.py:60]: Ok.

File: plugins/warns.py (hunk 6 above)
Comment (AndrielFR) [reply plugins/warns.py:39]: Yes, but we are not in the old bot, the new bot has come to improve, not to continue with the old things.

File: plugins/warns.py [hunk 8]
```diff
@@ -27,12 +27,6 @@ def add_warns(chat_id, user_id, number):
         db.commit()
     return True
 
-
-def set_warns_limit(chat_id, rules):
```
Comment (AndrielFR) [inline plugins/warns.py]: Good decision, will you add such a command later?

**Pull request merged**